src/ui/FacetsMobileMode/FacetsMobileMode.ts

```diff
@@ -0,0 +1,91 @@
+import { Component } from '../Base/Component';
+import { ComponentOptions } from '../Base/ComponentOptions';
+import { IComponentBindings } from '../Base/ComponentBindings';
+import { exportGlobally } from '../../GlobalExports';
+import { Utils } from '../../Core';
+
+export interface IFacetsMobileModeOptions {
+  breakpoint?: number;
+  isModal?: boolean;
+  showBackgroundWhileOpen?: boolean;
+  lockScroll?: boolean;
+  scrollContainer?: HTMLElement;
+}
+
+/**
+ * This component lets you customize the mobile responsive behavior of facets in your search interface.
+ *
+ * **Notes:**
+ * - You can include this component anywhere under the root element of your search interface.
+ * - You should only include this component once in your search interface.
+ * - If you do not include this component in your search interface, facets will still have a default mobile responsive behavior.
+ */
+export class FacetsMobileMode extends Component {
+  static ID = 'FacetsMobileMode';
+
+  /**
+   * @componentOptions
+   */
+  static options: IFacetsMobileModeOptions = {
+    /**
+     * The screen width (in number of pixels) at which facets should enter mobile responsive mode and be collapsed under a single button.
+     *
+     * **Default:** `800`
+     */
+    breakpoint: ComponentOptions.buildNumberOption(),
+    /**
+     * Whether to display the facets in a modal instead of a pop-up when the end user expands them in mobile responsive mode.
+     */
+    isModal: ComponentOptions.buildBooleanOption({ defaultValue: false }),
+    /**
+     * Whether to display a background behind the facets when the end user expands them in mobile responsive mode.
+     */
+    showBackgroundWhileOpen: ComponentOptions.buildBooleanOption({
+      postProcessing: (value, options) => (Utils.isNullOrUndefined(value) ? !options.isModal : value)
+    }),
+    /**
+     * Whether to disable scrolling in the search interface while facets are expanded in mobile responsive mode.
+     */
+    lockScroll: ComponentOptions.buildBooleanOption({ defaultValue: false }),
+    /**
+     * The element whose scrolling is locked.
```

Comment:
Suggestion:

> The HTML element whose vertical scrolling should be locked while facets are expanded in mobile responsive mode.
>
> By default, the component tries to detect and use the first ancestor element whose CSS `overflow-y` attribute is set to `scroll`, starting from the `FacetsMobileMode`'s element itself. If no such element is found, the `document.body` element is used.
>
> Since this heuristic is not perfect, we strongly recommend that you manually set this option by explicitly specifying the desired CSS selector.
>
> **Example:** `data-scroll-container-selector='#someCssSelector'`

Reply:
Thanks! I'm replacing the description with your suggestion.